libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -158,6 +158,20 @@ void DeclarationTypeChecker::endVisit(UserDefinedTypeName const& _typeName)
 		_typeName.annotation().type = TypeProvider::enumType(*enumDef);
 	else if (ContractDefinition const* contract = dynamic_cast<ContractDefinition const*>(declaration))
 		_typeName.annotation().type = TypeProvider::contract(*contract);
+	else if (auto userDefinedValueType = dynamic_cast<UserDefinedValueType const*>(declaration))
+	{
+		TypeName const* typeName = userDefinedValueType->typeName();
+		solAssert(typeName, "");
+		if (!dynamic_cast<ElementaryTypeName const*>(typeName))
```

Comment:
Don't we also want to restrict these to value types? (`string` and `bytes` are also elementary type names)

Reply:
This is done in the TypeChecker.

Reply:
But we could do it already here, can't we? Shouldn't we do it as early as possible? Otherwise we might get into trouble if we have a forward reference and the type checker somehow assumes that the underlying type is a value type.

Reply:
Sure, I can move it here.